browser/ui/views/infobars/brave_global_infobar_manager.cc

```diff
@@ -0,0 +1,192 @@
+          ->RemoveObserver(observer);
+      infobar_manager->RemoveInfoBar(current_infobar);
+    }
+  }
+}
+
+void RemoveObserverForInfobarManagers(
+    const infobars::InfoBarDelegate::InfoBarIdentifier& id,
+    BraveGlobalInfoBarManager* observer) {
+  for (const auto* browser : *BrowserList::GetInstance()) {
+    const auto* tab_strip_model = browser->tab_strip_model();
+    for (int i = 0; i < tab_strip_model->count(); ++i) {
+      auto* web_contents = tab_strip_model->GetWebContentsAt(i);
+      DCHECK(web_contents);
+      if (!web_contents) {
+        continue;
+      }
+      if (auto* infobar_manager =
+              infobars::ContentInfoBarManager::FromWebContents(web_contents)) {
+        infobar_manager->RemoveObserver(observer);
+        RemoveInfobarsByIdentifier(infobar_manager, id, observer);
+      }
+    }
+  }
+}
+}  // namespace
+
+BraveGlobalInfoBarManager::BraveGlobalInfoBarManager(
+    std::unique_ptr<BraveGlobalConfirmInfoBarDelegateFactory> delegate_factory)
+    : delegate_factory_(std::move(delegate_factory)) {}
+
+BraveGlobalInfoBarManager::~BraveGlobalInfoBarManager() {
+  RemoveObserverForInfobarManagers(delegate_factory_->GetInfoBarIdentifier(),
+                                   this);
+}
+
+void BraveGlobalInfoBarManager::Show() {
+  DCHECK(is_closed_);
+  if (!is_closed_) {
+    return;
+  }
+  is_closed_ = false;
+  browser_tab_strip_tracker_ =
+      std::make_unique<BrowserTabStripTracker>(this, nullptr);
+  browser_tab_strip_tracker_->Init();
+}
+
+void BraveGlobalInfoBarManager::TabChangedAt(content::WebContents* contents,
+                                             int index,
+                                             TabChangeType change_type) {
+  infobars::ContentInfoBarManager* infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(contents);
+  MaybeAddInfoBar(infobar_manager);
+}
+
+void BraveGlobalInfoBarManager::OnTabStripModelChanged(
+    TabStripModel* tab_strip_model,
+    const TabStripModelChange& change,
+    const TabStripSelectionChange& selection) {
+  if (!selection.new_contents || is_closed_) {
+    return;
+  }
+
+  infobars::ContentInfoBarManager* new_infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(selection.new_contents);
+  if (!new_infobar_manager) {
+    return;
+  }
+
+  MaybeAddInfoBar(new_infobar_manager);
+
+  if (!selection.old_contents || !selection.active_tab_changed()) {
+    return;
+  }
+
+  infobars::ContentInfoBarManager* old_infobar_manager =
+      infobars::ContentInfoBarManager::FromWebContents(selection.old_contents);
+
+  if (!old_infobar_manager) {
+    return;
+  }
+  old_infobar_manager->RemoveObserver(this);
+  RemoveInfobarsByIdentifier(old_infobar_manager,
+                             delegate_factory_->GetInfoBarIdentifier(), this);
+}
+
+void BraveGlobalInfoBarManager::MaybeAddInfoBar(
+    infobars::InfoBarManager* infobar_manager) {
+  DCHECK(infobar_manager);
```

Comment:
Please also check Incognito behavior. Infobar shouldn't be shown here

Reply:
fixed